cellSAM/cellsam_pipeline.py

```diff
@@ -79,6 +79,70 @@ def cellsam_pipeline(
         iou_threshold=0.5,
         filter_below_min=True, 
 ):
+    """Run the cellsam inference pipeline on `img`.
+
+    Cellsam is capable of segmenting a variety of cells (bacteria,
+    eukaryotic, etc.) spanning all forms of microscopy (brightfield,
+    phase, autofluorescence, electron microscopy) and
+    staining (H&E, PAS, etc.) / multiplexed (codex, mibi, etc.)
+    modalities.
+
+    Parameters
+    ----------
+    img : array_like with shape ``(W, H)`` or ``(W, H, C)``, where C is 1 or 3
+        The image to be segmented. For multiple-channel images, `img` should
+        have the following format:
+
+          - **Stained images (e.g H&E)**: ``(W, H, C)`` where ``C == 3``
+            representing color channels in RGB format.
+          - **Multiplexed images**: ``(W, H, C)`` where ``C == 3`` and the
+            channel ordering is: ``(blank, nuclear, membrane)``. The
+            ``membrane`` channel is optional, in which case a nuclear segmentation
+            is returned.
+    chunks : int
+        TODO: should this be an option?
+    model_path : str or pathlib.Path, optional
+        Path to the model weights. If `None` (the default), the latest released
+        cellsam generalist model is used.
+
+        .. note:: Downloading the model requires internet access
+
+    bbox_threshold : float in range [0, 1], default=0.4
+        TODO: Detail effects of this parameter
+    low_contrast_enhancement : bool, default=True
+        TODO: Detail effects of this parameter
+    swap_channels : bool, default=False
+        TODO: this should be removed with loading from file
+    use_wsi : bool, default=True
+        Whether to use tiling to support large images, default is True.
+        Generally, tiling is not required when there are fewer than ~3000
+        cells in an image.
+    gauge_cell_size : bool, default=True
+        TODO: Detail effects of this parameter
```

Comment:
Similar to `low_contrast_enhancement` - what does this do and when is it useful?

Reply:
If the cells are really small, it may make sense to re-segment with tiling instead of vanilla

Reply:
Same deal here - can you propose a parameter description?